kbncontent.go

```diff
@@ -0,0 +1,319 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// Package kbncontent implements routines for analyzing Kibana content.
+//
+// It provides information about Kibana assets and a single source of truth for what things are legacy and should no longer be used.
+package kbncontent
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/stretchr/objx"
+)
+
+// Describes a visualization.
+// In this case, "visualization" means anything which can be embedded in a dashboard.
+type VisualizationDescriptor struct {
+	Doc    map[string]interface{}
+	SoType string
+	Link   string
+}
+
+func (v VisualizationDescriptor) tryDocumentPaths(paths []string) string {
+	m := objx.Map(v.Doc)
+
+	for _, path := range paths {
+		if m.Get(path).IsStr() {
+			return m.Get(path).Str()
+		}
+	}
+
+	return ""
+}
+
+// root-level visualization type
+// currently empty for Lens
```

Comment:
And for example here:
```suggestion
// Type returns the root root-level visualization type of the visualization.
// It is currently empty for Lens.
```